src/main/resources/META-INF/resources/primefaces/columntoggler/columntoggler.js

```diff
@@ -397,6 +398,38 @@ PrimeFaces.widget.ColumnToggler = PrimeFaces.widget.DeferredWidget.extend({
         this.changeTogglerState(column, false);
         this.fireToggleEvent(false, (index - 1));
         this.updateColspan();
+        this.resizeColumns(false);
+    },
+
+    /**
+     * Resize columns after toggle
+     * @param {boolean} isChecked `true` when the number of column will increase
+     * 'false' when they are decrease.
+     */
+    resizeColumns: function(isChecked) {
+
+        var columns = this.table.find('> .ui-datatable-tablewrapper > table > thead > tr > th');
+
+        if(columns.length) {
+            var visibleColumnLength = columns.filter(':visible').length;
+
+            if (isChecked && visibleColumnLength > 1) {
+                columns.each(function() {
+                    var col = $(this),
+                        colStyle = col[0].style,
```

Comment:
I haven't checked by can `colStyle` be NULL or undefined if a column has no style?  Then the next line will throw an error when its checks its width.

Reply:
Columns always have style from ui-datatable or themes. In addition, the same lines are used exactly in 5 different places, even in the datatable.js file.